File: backend/Entities/Badge.cs
using backend.Enums;

namespace backend.Entities;

public class Badge
{
    public Guid Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public string IconUrl { get; set; } = string.Empty;

    public BadgeType Type { get; set; }
}

File: frontend/src/features/badges/types/badge.ts
export type BadgeType =
  | "Completion"
  | "Distance"
  | "Region"
  | "Difficulty"
  | "Streak";

export type RawBadgeType = BadgeType | number;

export interface BadgeResponse {
  id: string;
  name: string;
  description: string;
  iconUrl: string;
  type: BadgeType;
  isUnlocked: boolean;
  unlockedAt: string | null;
  currentValue: number;
  targetValue: number;
  progressLabel: string;
}

export interface RawBadgeResponse {
  id: string;
  name: string;
  description: string;
  iconUrl: string;
  type: RawBadgeType;
  isUnlocked: boolean;
  unlockedAt: string | null;
  currentValue: number;
  targetValue: number;
  progressLabel: string;
}
